### PR TITLE
Allow SDK to be built with unknown traits in models (upgrading Smithy to 1.29.0 as part of it)

### PR DESCRIPTION
**Pull request by ysaito1001**
## Motivation and Context
Addresses https://github.com/awslabs/smithy-rs/issues/2104

## Description
This PR allows us to build `aws-sdk-rust` even with unknown traits present in the AWS models. This requires `Smithy` 1.29.0 that includes [support](https://github.com/awslabs/smithy/pull/1685) for an extension property, [allowUnknownTraits](https://smithy.io/2.0/guides/building-models/gradle-plugin.html?highlight=allowunknowntraits#smithy-extension-properties).

## Testing
Manually added a unknown trait into an AWS model like so:
```
diff --git a/aws/sdk/aws-models/sts.json b/aws/sdk/aws-models/sts.json
--- a/aws/sdk/aws-models/sts.json
+++ b/aws/sdk/aws-models/sts.json
@@ -69,6 +69,7 @@
                 "aws.auth#sigv4": {
                     "name": "sts"
                 },
+                "aws.protocols#awsQuery2": {},
                 "aws.protocols#awsQuery": {},
```
and confirmed `/gradlew :aws:sdk:assemble` built SDK successfully. Also verified that without `allowUnknownTraits = true` in `aws/sdk/build.gradle.kts`, `/gradlew :aws:sdk:assemble` failed with the above unknown trait.

----

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

**Comment by jdisanti** [comment]
Looks good to me. Holding off on approval until the Smithy upgrade issues are resolved.

**Review by jdisanti — APPROVED** [review]

**Review by david-perez — APPROVED** [review]
I think the PR title and commit title of this should mention the fact we're updating to Smithy 1.29.0.

File: codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/protocol/ServerProtocolTestGenerator.kt [hunk 1]
```diff
@@ -828,6 +828,8 @@ class ServerProtocolTestGenerator(
             FailingTest(RestJsonValidation, "RestJsonMalformedEnumMapValue_case1", TestType.MalformedRequest),
             FailingTest(RestJsonValidation, "RestJsonMalformedEnumString_case0", TestType.MalformedRequest),
             FailingTest(RestJsonValidation, "RestJsonMalformedEnumString_case1", TestType.MalformedRequest),
+            FailingTest(RestJsonValidation, "RestJsonMalformedEnumTraitString_case0", TestType.MalformedRequest),
+            FailingTest(RestJsonValidation, "RestJsonMalformedEnumTraitString_case1", TestType.MalformedRequest),
```
[the remark this answers is not in the document]
Comment (david-perez) [reply codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/protocol/ServerProtocolTestGenerator.kt:832]: @crisidev See https://github.com/awslabs/smithy/issues/1737.

[the remark this answers is not in the document]
Comment (ysaito1001) [reply codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/protocol/ServerProtocolTestGenerator.kt:832]: Julian and I originally suspected that these two were related to the TODO comment above (hence part of #2472). If these newly failed tests are more closely related to #1737, I'll move them from this group, place them in new space, and separately link them to #1737.

[the remark this answers is not in the document]
Comment (ysaito1001) [reply codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/protocol/ServerProtocolTestGenerator.kt]: Addressed in 5626f56

**Pull request merged**